src/context/chains.ts

```diff
@@ -1,29 +1,39 @@
 import { z } from 'zod';
 
-import { ChainMap, ChainMetadata, ChainMetadataSchema, chainMetadata } from '@hyperlane-xyz/sdk';
+import { GithubRegistry, chainMetadata } from '@hyperlane-xyz/registry';
+import { ChainMap, ChainMetadata, ChainMetadataSchema } from '@hyperlane-xyz/sdk';
 
-import ChainsJson from '../consts/chains.json';
 import { chains as ChainsTS } from '../consts/chains.ts';
 import ChainsYaml from '../consts/chains.yaml';
+import { config } from '../consts/config.ts';
 import { cosmosDefaultChain } from '../features/chains/cosmosDefault';
 import { logger } from '../utils/logger';
 
-export const ChainConfigSchema = z.record(
-  ChainMetadataSchema.and(z.object({ mailbox: z.string().optional() })),
-);
-
-export function getChainConfigs() {
+export async function assembleChainMetadata() {
   // Chains must include a cosmos chain or CosmosKit throws errors
-  const result = ChainConfigSchema.safeParse({
+  const result = z.record(ChainMetadataSchema).safeParse({
     cosmoshub: cosmosDefaultChain,
-    ...ChainsJson,
     ...ChainsYaml,
     ...ChainsTS,
   });
   if (!result.success) {
     logger.warn('Invalid chain config', result.error);
     throw new Error(`Invalid chain config: ${result.error.toString()}`);
   }
-  const customChainConfigs = result.data as ChainMap<ChainMetadata & { mailbox?: Address }>;
-  return { ...chainMetadata, ...customChainConfigs };
+  const customChainMetadata = result.data as ChainMap<ChainMetadata>;
+
+  const registry = new GithubRegistry({ uri: config.registryUrl });
+  let defaultChainMetadata = chainMetadata;
+  if (config.registryUrl) {
+    logger.debug('Using custom registry', config.registryUrl);
+    defaultChainMetadata = await registry.getMetadata();
+  } else {
+    logger.debug('Using default published registry');
+    // Note: this is an optional optimization to pre-fetch the content list
+    // and avoid repeated request from the chain logos that will use this info
+    await registry.listRegistryContent();
+  }
```

Comment:
any way we can do this at build time?

Reply:
We could yes. But if we're looking to make this faster, I would start with this: https://github.com/hyperlane-xyz/hyperlane-registry/issues/29#issuecomment-2107820979